File: app/src/main/java/com/example/nickolas/vk/models/remote/DialogsDataSource.java
package com.example.nickolas.vk.models.remote;

import com.example.nickolas.vk.api.VkApi;

import okhttp3.ResponseBody;
import rx.Observable;

/**
 * Created by Nickolas on 01.10.2017.
 */

public class DialogsDataSource implements IDialogsDataSource {

    private VkApi api;

    public DialogsDataSource(VkApi api){
        this.api = api;
    }

    @Override
    public Observable<ResponseBody> getDialogs(int offset, String token) {
        return api.getDialogs(offset, token, "5.68");
    }
}

File: app/src/main/java/com/example/nickolas/vk/models/remote/DialogsDataSource.kt
package com.example.nickolas.vk.models.remote

import com.example.nickolas.vk.api.VkApi
import com.example.nickolas.vk.models.enteties.Dialog
import com.example.nickolas.vk.utils.GetIdsFrom

import okhttp3.ResponseBody
import rx.Observable


class DialogsDataSource(private val api: VkApi) : IDialogsDataSource {

    override fun getDialogs(offset: Int, token: String): Observable<ResponseBody>
            = api.getDialogs(offset, token)

    override fun getUsersData(dialogs: List<Dialog>): Observable<ResponseBody>
            = api.usersGet(GetIdsFrom.dialogList(dialogs))

}
